e2e/files.spec.ts

```diff
@@ -0,0 +1,283 @@
+          name: firstUserName,
+        }),
+      ).toBeVisible()
+    },
+  )
+
+  await testAuth.step(
+    `Second user is ${
+      secondUserVisible ? '' : 'not'
+    } listed in File Sharing settings`,
+    async () => {
+      const secondUserCell = firstUserPage.getByRole('cell', {
+        name: secondUserName,
+      })
+      if (secondUserVisible) {
+        await expect(secondUserCell).toBeVisible()
+      } else {
+        await expect(secondUserCell).not.toBeVisible()
+      }
+    },
+  )
+}
+
+const confirmAndClosePermissionsSavedAlert = async (page: Page) => {
+  const permissionsAlert = page.getByRole('alert').filter({
+    has: page.getByText('Permissions were successfully saved to Synapse'),
+  })
+  await expect(permissionsAlert).toBeVisible()
+  await permissionsAlert.getByRole('button').click()
+  await expect(permissionsAlert).not.toBeVisible()
+}
+
+const saveFileSharingSettings = async (page: Page) => {
+  const saveButton = page.getByRole('button', { name: 'Save' })
+  await saveButton.click()
+  await expect(saveButton).not.toBeVisible()
+
+  // Modal is closed after settings are saved
+  await expect(
+    page.getByRole('heading', { name: 'File Sharing Settings' }),
+  ).not.toBeVisible()
+
+  await confirmAndClosePermissionsSavedAlert(page)
+}
+
+let userProject: Project
+let fileHandleIds: string[] = []
+
+test.describe('Files', () => {
+  testAuth.beforeAll(async ({ browser, storageStatePaths }) => {
+    const userContext = await browser.newContext({
+      storageState: storageStatePaths['swc-e2e-user'],
+    })
+    const userPage = await userContext.newPage()
+    const userAccessToken = await getAccessTokenFromCookie(userPage)
+
+    // create project
+    const userProjectName = generateEntityName('project')
+    const userProjectId = await createProject(
+      userProjectName,
+      userAccessToken,
+      userPage,
+    )
+    userProject = { name: userProjectName, id: userProjectId }
+
+    await userContext.close()
+  })
+
+  testAuth.afterAll(async ({ browser }) => {
+    const context = await browser.newContext()
+    const page = await context.newPage()
+    const accessToken = getAdminPAT()
+
+    // delete project
+    if (userProject.id) {
+      await deleteProject(userProject.id, accessToken, page)
+    }
+
+    // delete fileHandles
+    for await (const fileHandleId of fileHandleIds) {
+      await deleteFileHandleWithRetry(accessToken, fileHandleId, page)
+    }
+
+    await context.close()
+  })
+
+  testAuth(
+    'File sharing',
+    async ({ userPage, validatedUserPage, browserName }, testInfo) => {
+      if (browserName === 'webkit') {
+        test.info().annotations.push({
+          type: 'very slow',
+          description: `webkit in CI only. May be related to the InviteWidget. 
+            Evaluate if timeout can be removed after addressing 
+            https://sagebionetworks.jira.com/browse/SWC-6569.`,
+        })
+        test.setTimeout(testInfo.timeout * 5)
+      }
```

Comment:
The files test is intermittently very slow in webkit in CI (5-7 minutes in CI vs 35 seconds locally), but does eventually pass. Increase overall test timeout and timeout of particularly slow expectations to help this test pass when it is slow.

Reply:
That's really weird. This annotations system is cool though, so thanks for adding all of this potentially helpful information.